### PR TITLE
gnome-extra/nautilus-dropbox: Version Bump to 2019.02.14

### PR DESCRIPTION
**Pull request by j-licht**
Bug: https://bugs.gentoo.org/546024
Bug: https://bugs.gentoo.org/706486
Package-Manager: Portage-2.3.84, Repoman-2.3.20
Signed-off-by: Jonas Licht <jonas.licht@fem.tu-ilmenau.de>

### CODE REVIEW

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
```
Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: 7 should be available here.

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+GNOME2_LA_PUNT="yes"
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit autotools python-single-r1 gnome2
```
Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: I didn't see any .schema files being installed / handled, maybe *gnome2* is unneeded here?

Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: I didn't see any .py files / python scripts being installed either. Maybe you want to use *python-any-r1* instead? How is python being used in this?

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+RDEPEND="${PYTHON_DEPS}
+	gnome-base/nautilus
+	net-misc/dropbox
+	>=dev-libs/glib-2.14
+	$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}] ')
+	"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: Note that this is a BDEPEND with EAPI-7.

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+src_prepare() {
+	gnome2_src_prepare
+
+	# use system dropbox
+	sed \
+		-e "s|~/[.]dropbox-dist|${EPREFIX}/opt/dropbox|" \
+		-e "s|\(DROPBOXD_PATH = \).*|\1\"${EPREFIX}/opt/dropbox/dropboxd\"|" \
+		-i dropbox.in || die
+	# use system rst2man
+	epatch "${FILESDIR}"/${PN}-2019-system-rst2man.patch
```
Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: `epatch` is banned from being used in ebuilds, use `eapply` or global `PATCHES=()` array instead.

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	$(use_enable debug) \
+	--disable-static
+}
+
+src_install() {
+	gnome2_src_install
+
+	# removes files which conflicts with system dropbox
+	rm -r "${D}usr/share/applications"
+	rm -r "${D}usr/bin"
```
Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: Both need to be appended with `|| die`

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild (hunk 1 above)
Comment (ConiKost) [reply gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: Eclass `gnome2` does not support EAPI=7 :-(

Comment (a17r) [reply gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: ...which is solved by dropping it :p

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild (hunk 2 above)
Comment (j-licht) [reply gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: nautilus-dropbox needs dev-python/pygobject also known as gi at runtime

File: gnome-extra/nautilus-dropbox/metadata.xml [hunk 6]
```diff
@@ -1,5 +1,15 @@
...
+	<maintainer type="person">
+		<name>Jonas Licht</name>
+		<email>jonas.licht@gmail.com</email>
+	</maintainer>
+	<maintainer type="project">
+		<email>proxy-maint@gentoo.org</email>
+		<name>Proxy Maintainers</name>
+	</maintainer>
+	<longdescription lang="en">
+			A extension of nautilus to display the sync state of an Dropbox folder and add a right click menu to mange files in the Dropbox folder.
```
Comment (a17r) [inline gnome-extra/nautilus-dropbox/metadata.xml]: please fix indendation

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+		-e "s|~/[.]dropbox-dist|${EPREFIX}/opt/dropbox|" \
+		-e "s|\(DROPBOXD_PATH = \).*|\1\"${EPREFIX}/opt/dropbox/dropboxd\"|" \
+		-i dropbox.in || die
+	AT_NOELIBTOOLIZE=yes eautoreconf
+}
+
+src_configure() {
+	econf \
+	$(use_enable debug) \
+	--disable-static
```
Comment (a17r) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: please fix indendation

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+src_install() {
+	default
+
+	# removes files which conflicts with system dropbox
+	rm -r "${D}/usr/share/applications" || die
+	rm -r "${D}/usr/bin" || die
+}
+
+pkg_postinst() { xdg_icon_cache_update; }
+pkg_postrm() { xdg_icon_cache_update; }
```
Comment (a17r) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild]: please put the function call in a separate line so we keep a uniform ebuild style.

**Comment by a17r** [comment]
lgtm purely by looking at the ebuild, @juippis what do you think?

File: gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+RDEPEND="${PYTHON_DEPS}
+	gnome-base/nautilus
+	net-misc/dropbox
+	>=dev-libs/glib-2.14
+	$(python_gen_cond_dep 'dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]' )"
+
+DEPEND="${RDEPEND}"
+
+BDEPEND="
+	$(python_gen_cond_dep 'dev-python/docutils[${PYTHON_MULTI_USEDEP}]' )
```
Comment (juippis) [inline gnome-extra/nautilus-dropbox/nautilus-dropbox-2019.02.14.ebuild:30]: You don't need/want the `python_gen_cond_dep` on a BDEPEND, your CHOST and CBUILD can have different default python interpreter.

**Comment by juippis** [comment]
Let this stay in the tree for ~10-14 days and if no major bug reports arise, then stabilize 2019.02.14 and remove the older version.

**Pull request closed without merging**